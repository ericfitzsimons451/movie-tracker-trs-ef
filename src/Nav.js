import React from 'react'
import { NavLink } from 'react-router-dom'

const Nav = () => {
    return (
        <div>
            <NavLink to='/' className="movies-link">Show Movies</NavLink>
            <NavLink to='/favorites' className="favorites-link">Show Favorites</NavLink>
        </div>
    )
}

export default Nav