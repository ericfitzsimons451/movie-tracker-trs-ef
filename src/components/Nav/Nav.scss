.nav-container {
  display: flex;
  justify-content: flex-end;
  align-items: center;
  text-align: right;
}

.nav-link {
  font-size: 1rem;
  color: white;
  text-decoration: none;
  cursor: pointer;
  width: 100px;
  &:hover {
    color: #db0000;
  }
}
