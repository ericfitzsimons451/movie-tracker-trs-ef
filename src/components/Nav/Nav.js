import React, { Component } from "react";
import "./Nav.scss";
import { NavLink, withRouter } from "react-router-dom";
import { connect } from "react-redux";
import { PropTypes } from "prop-types";
import { loginUser, setErrorMessage, storeFavorites } from "../../actions";
import allPurposeFetch from "../../allPurposeFetch";

const Nav = props => {
  console.log(props)
  const signOut = e => {
    const data = {
      name: "",
      email: ""
    };
    props.loginUser(data);
  };

  const validateLogin = async e => {
    const { history } = props;
    if (!props.user.email) {
      props.setErrorMessage("You must be signed in to view favorites");
      history.push("/login");
    } else {
      await fetchFavoriteMovies();
      await history.push("/favorites");
    }
  };

  const fetchFavoriteMovies = async () => {
    const url = `http://localhost:3000/api/users/${props.user.id}/favorites`;
    const favorites = await allPurposeFetch(url);
    const favs = filterFavorites(favorites.data);
    props.storeFavorites(favs);
  };

  const filterFavorites = array => {
    const favs = [];
    props.movies.forEach(movie => {
      array.forEach(favorite => {
        if (favorite.movie_id === movie.id) {
          favs.push(movie);
        }
      });
    });
    return favs;
  };

  let welcomeMessage;
  let authLink;
  if (!props.user.email) {
    authLink = (
      <NavLink to="/login" className="nav-link">
        Sign In
      </NavLink>
    );
  } else {
    welcomeMessage = <h3 className="nav-link">Welcome, {props.user.name}!</h3>;
    authLink = (
      <NavLink to="/" className="nav-link" onClick={signOut}>
        Sign Out
      </NavLink>
    );
  }

  return (
    <div className="nav-container">
      <NavLink to="/" className="nav-link">
        Movies
      </NavLink>
      <button type="submit" className="nav-link" onClick={validateLogin}>
        Favorites
      </button>
      {authLink}
      {welcomeMessage}
    </div>
  );
};
export const mapStateToProps = state => ({
  user: state.user,
  favorites: state.favorites,
  movies: state.movies
});

export const mapDispatchToProps = dispatch => ({
  loginUser: data => dispatch(loginUser(data)),
  setErrorMessage: message => dispatch(setErrorMessage(message)),
  storeFavorites: favorites => dispatch(storeFavorites(favorites))
});

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(Nav)
);

Nav.propTypes = {
  user: PropTypes.object,
  favorites: PropTypes.array,
  dispatch: PropTypes.func
};
