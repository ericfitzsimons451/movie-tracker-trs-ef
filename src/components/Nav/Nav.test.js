import React from 'react'
import { shallow } from 'enzyme'
import Nav from './Nav'

describe('Nav', () => {
    let wrapper
    it('should match the snapshot with all data passed down', () => {
        const props = {
            props: 'props'
        }

//obviously, this is just to show that this is not a great
//way to test a component.  Flesh out props better, but know
//that even though we are getting the props from mapSTP, we can
//we can call them anything that we want in the test file.

        wrapper = shallow(<Nav props={props}/>)
        expect(wrapper).toMatchSnapshot()
    })
    describe('signOut', () => {

    })

    describe('filterFavorites', () => {

    })

    describe('mapStateToProps', () => {

    })

    describe('matchDispatchToProps', () => {

    })
})