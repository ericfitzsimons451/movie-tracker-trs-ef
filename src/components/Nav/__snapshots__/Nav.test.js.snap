// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot with all data passed down 1`] = `
<Route>
  <Component />
</Route>
`;
